New Tools package, carved off from util package

DanVanAtta opened the pull request:
Move the map making tools to their own package tree. The updated package structure is in the screen shot below. Most of the existing classes were plucked out of 'util' and moved to 'tools.image' or 'tools.map.making'. The XML creator was moved to 'tools.map.xml.creator'

![package_restructure](https://cloud.githubusercontent.com/assets/12397753/15565038/7ef94d18-22cc-11e6-8bba-f64e727fda32.png)

I went through the package imports of the tools classes to try and pull stuff in that was self contained. We still have references to a few other classes, but they look to be shared. There is on Swing UI component moved over, which will be nice to have distinctly out of the main client source code.

* Notes on what has change: For the most part everything is a straight file move, except:
- (important) replaced invocation of tool classes by name (as a string value) with a compile time checked method call: `class.getName()`
- (minor) replaced a wildcard import with single imports
- (minor) fixed a typo

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/triplea-game/triplea/759)
<!-- Reviewable:end -->

ron-murhammer commented:
@DanVanAtta Looks like this needs rebased.

DanVanAtta commented:
@ron-murhammer  - pls have a look at the contents of this change. The merge conflicts with this one could be error prone to resolve, and @frigoref  will likely want to integrate with these changes ASAP.

DanVanAtta commented:
@frigoref  - heads up on the xml creator being moved, have a look here when you get the chance

ron-murhammer commented on src/games/strategy/triplea/ui/MapData.java:
```diff
@@ -31,7 +31,7 @@
 import games.strategy.triplea.image.UnitImageFactory;
 import games.strategy.ui.Util;
-import games.strategy.util.PointFileReaderWriter;
+import tools.map.making.PointFileReaderWriter;
```
Thoughts on MapData importing from tools package? Any clean way to avoid that?

DanVanAtta replied:
It's a good catch, we should not have standard tripleA code calling into the tools packages. Some options:
- split PointFileReaderWriter up in to two,  core code vs non-core.
- move PointFileReaderWriter back ot the core code section, `games.strategy.util`

ron-murhammer replied:
I'd say for this PR just move it back to core. Can always do a separate PR to split it up.

DanVanAtta replied:
Sounds good, stay tuned for the update.

ron-murhammer commented on src/tools/map/making/PointFileReaderWriter.java:
```diff
@@ -1,4 +1,4 @@
-package games.strategy.util;
+package tools.map.making;
```
Maybe better to leave this one in games/strategy for now since its used by GameData.

ron-murhammer commented:
@DanVanAtta I like the changes and direction this is going. Just one class we might want to leave where it was on.

I'd like us to consider actually going further and before the next release maybe rename all the packages. I think 'games.strategy' is kind of dumb. The root package of the project should probably be 'triplea'. So you'd have something like:
triplea.ai
triplea.common
triplea.engine
triplea.rules
triplea.tools
triplea.ui

DanVanAtta commented:
Let's discuss packages further in: https://github.com/triplea-game/triplea/issues/763

About releasing:
I don't think we will necessarily freeze the code. At some time we will decide we are ready to do release testing. The latest build at that time becomes the release candidate.  Either:
- things look good and we flip the 'prerelease' checkbox to 'release'
- we add a bugfix to the latest master, and repeat the release testing on the latest build

We will have some control over how much new stuff enters the pipeline once we start doing release testing, but we should still continue incorporating new changes. That set of new changes should be pretty small, so we will test that along with everything else there is to test.. 

For this PR: I need to do some work on it still

DanVanAtta commented:
updated, moved PointFileReaderWriter back to core packages.

ron-murhammer commented:
@DanVanAtta Looks like this needs rebased.

DanVanAtta commented:
updated

DanVanAtta commented:
@ron-murhammer is this good to go? Seems like it's in the right direction. Let's tackle more package re-org in multiple future PRs.

DanVanAtta commented:
Now has merge conflicts.. It's important we get the tools fixes in and the classes moved. From there it is much easier to do future work.

DanVanAtta commented:
updated.
to be clear, for this PR I would like to keep the scope limited to what is here. Let's use this as a stepping stone for the next step. Before we do anything else we need to do some careful work to make sure any package moves will be safe. Note that the tools class and package names were hardcoded, and fixed in this PR so that we could move the tools safely.
